### PR TITLE
Migrate sub to use the new k8s clients

### PR DESCRIPTION
**Pull request by stevekuznetsov**
Signed-off-by: Steve Kuznetsov skuznets@redhat.com

Depends on #11060

/assign @cjwagner @fejta @munnerz
/cc @sebastienvas @BenTheElder @krzyzacy

### CODE REVIEW

**Comment by fejta** [comment]
Please recreate this PR (and all the others) without the first two commits.

**Comment by krzyzacy** [comment]
rebase the announcements :-)

File: prow/cmd/sub/main.go [hunk 1]
```diff
@@ -110,14 +109,13 @@ func main() {
 		tokenGenerator = secretAgent.GetTokenGenerator(flagOptions.pushSecretFile)
 	}
 
-	prowjobClient, err := flagOptions.client.ProwJobClient()
+	prowjobClient, err := flagOptions.client.ProwJobClient(configAgent.Config().ProwJobNamespace, flagOptions.dryRun)
```
Comment (fejta) [inline prow/cmd/sub/main.go]: `// TODO(fejta): use the currently configured prowjobnamespace :)`

**Comment by sebastienvas** [comment]
/lgtm
thanks!

**Comment by fejta** [comment]
needs more
/lgtm

**Pull request merged**